### PR TITLE
docs(samples): add template/monitoring samples

### PR DESCRIPTION
**Pull request by FrodoTheTrue**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-security-private-ca/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by leahecole** [comment]
Hey @savijatv! Sorry I haven't sent you any reviews recently - can you PTAL at this and let me know when you've reviewed it?

File: samples/snippets/delete_certificate_template.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+def delete_certificate_template(
+    project_id: str, location: str, certificate_template_id: str,
+) -> None:
+    """
+    Delete the certificate template present in the given project and location.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
```
[the remark this answers is not in the document]
Comment (leahecole) [reply samples/snippets/delete_certificate_template.py:28]: great catch - they are different entities

File: samples/snippets/list_certificate_templates.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# [START privateca_list_certificate_template]
+import google.cloud.security.privateca_v1 as privateca_v1
+
+
+def list_certificate_templates(project_id: str, location: str) -> None:
+    """
+    List the certificate templates present in the given project and location.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
```
[the remark this answers is not in the document]
Comment (leahecole) [reply samples/snippets/list_certificate_templates.py:26]: great catch - they are different entities

File: samples/snippets/monitor_certificate_authority.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+# [START privateca_monitor_ca_expiry]
+import google.cloud.monitoring_v3 as monitoring_v3
+
+
+def create_ca_monitor_policy(project_id: str) -> None:
+    """
+    Create a monitoring policy that notifies you 30 days before a managed CA expires.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
```
[the remark this answers is not in the document]
Comment (leahecole) [reply samples/snippets/monitor_certificate_authority.py:26]: great catch - they are different entities

[the remark this answers is not in the document]
Comment (FrodoTheTrue) [reply samples/snippets/monitor_certificate_authority.py:26]: For python-compute we use the same description: https://github.com/googleapis/python-compute/blob/main/samples/snippets/sample_create_vm.py#L167

Here we accept both entities as an input parameter: project ID and project number

File: samples/snippets/monitor_certificate_authority.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        combiner=monitoring_v3.AlertPolicy.ConditionCombinerType.AND,
+        notification_channels=[channel.name],
+    )
+
+    policy = alertPolicyServiceClient.create_alert_policy(
+        name=notificationChannelServiceClient.common_project_path(project_id),
+        alert_policy=alert_policy,
+    )
+
+    print("Monitoring policy successfully created !", policy.name)
```
[the remark this answers is not in the document]
Comment (FrodoTheTrue) [reply samples/snippets/monitor_certificate_authority.py:74]: There is a good question. But this PR is just a python equivalent of Java samples: https://github.com/googleapis/java-security-private-ca/blob/main/samples/snippets/cloud-client/src/main/java/privateca/MonitorCertificateAuthority.java#L87 

We can fix it later I guess

File: samples/snippets/monitor_certificate_authority.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (savijatv) [reply samples/snippets/monitor_certificate_authority.py:26]: I still think that they're different. The Project ID starts with a letter whereas Project number is all numeric. See https://cloud.google.com/resource-manager/docs/creating-managing-projects.
Does the code accept both?

File: samples/snippets/delete_certificate_template.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rsamborski) [reply samples/snippets/delete_certificate_template.py:28]: I have double checked with a `gloud projects describe` and it seems that both project ID and number work in API calls. See more here: https://cloud.google.com/resource-manager/docs/creating-managing-projects

To be on the safe side we should test this sample with ID and number as well.

[the remark this answers is not in the document]
Comment (FrodoTheTrue) [reply samples/snippets/delete_certificate_template.py:28]: Agreed that this is a bit strange behavior
Makes sense to add a test.

File: samples/snippets/monitor_certificate_authority.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrodoTheTrue) [reply samples/snippets/monitor_certificate_authority.py:26]: Yep, have just checked, the code accepts both

File: samples/snippets/list_certificate_templates.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        location: location you want to use. For a list of locations, see: https://cloud.google.com/certificate-authority-service/docs/locations.
+    """
+
+    caServiceClient = privateca_v1.CertificateAuthorityServiceClient()
+
+    # List Templates Request.
+    request = privateca_v1.ListCertificateTemplatesRequest(
+        parent=caServiceClient.common_location_path(project_id, location),
+    )
+
```
[the remark this answers is not in the document]
Comment (FrodoTheTrue) [reply samples/snippets/list_certificate_templates.py:36]: I guess here we have either exception itself or printed list of templates

**Review by savijatv — APPROVED** [review]
If the code accepts both project ID and project number as the argument, please ignore my comment.

**Review by leahecole — APPROVED** [review]

**Pull request merged**